src/CattyTests/Project/ProjectMigratorTest.swift

```diff
@@ -0,0 +1,147 @@
+
+        let migrate = ProjectMigrator(fileManager: fileManager)
+
+        do {
+            try migrate.migrateToScene(project: project)
+        } catch {
+            XCTFail("Some error occured")
+        }
+
+        XCTAssertFalse(fileManager.fileExists(automaticScreenShortAtPath))
+        XCTAssertFalse(fileManager.directoryExists(imageDirectoryAtPath))
+        XCTAssertFalse(fileManager.directoryExists(soundDirectoryAtPath))
+        XCTAssertFalse(fileManager.fileExists(manualScreenShotAtPath))
+        XCTAssertFalse(fileManager.fileExists(screeShotAtPath))
+
+        automaticScreenShortAtPath = "\(String(describing: projectPath) + newFolderName) 1/automatic_screenshot.png"
+        imageDirectoryAtPath = "\(String(describing: projectPath) + newFolderName) 1/images"
+        soundDirectoryAtPath = "\(String(describing: projectPath) + newFolderName) 1/sounds"
+        manualScreenShotAtPath = "\(String(describing: projectPath) + newFolderName) 1/manual_screenshot.png"
+        screeShotAtPath = "\(String(describing: projectPath) + newFolderName) 1/screenshot.png"
+
+        XCTAssertTrue(fileManager.fileExists(automaticScreenShortAtPath))
+        XCTAssertTrue(fileManager.directoryExists(imageDirectoryAtPath))
+        XCTAssertTrue(fileManager.directoryExists(soundDirectoryAtPath))
+        XCTAssertTrue(fileManager.fileExists(manualScreenShotAtPath))
+        XCTAssertTrue(fileManager.fileExists(screeShotAtPath))
+    }
+
+    func testMigrateToSceneForInvalidCatrobatLanguageVersion() {
+        header.catrobatLanguageVersion = "0.991"
+        project.header = header
+
+        let fileManager = CBFileManagerMock(filePath: [] as! [String], directoryPath: [] as! [String])
+
+        let migrate = ProjectMigrator(fileManager: fileManager)
+
+        expect { try migrate.migrateToScene(project: self.project) }.to(throwError(ProjectMigratorError.unsupportedCatrobatLanguageVersion))
+    }
+
+    func testMigrateToSceneForProjectPathNotFound() {
+        let fileManager = CBFileManagerMock(filePath: [] as! [String], directoryPath: [] as! [String])
+
+        let migrate = ProjectMigrator(fileManager: fileManager)
+
+        expect { try migrate.migrateToScene(project: self.project) }.to(throwError(ProjectMigratorError.pathNotFound))
+    }
+
+    func testMigrateToSceneForFilePathNotFound() {
+        let projectPath = project.projectPath()
+
+        let imageDirectoryAtPath = "\(String(describing: project.projectPath()))images"
+        let soundDirectoryAtPath = "\(String(describing: project.projectPath()))sounds"
+
+        let fileManager = CBFileManagerMock(filePath: [] as! [String], directoryPath: [imageDirectoryAtPath, soundDirectoryAtPath, projectPath])
+
+        let migrate = ProjectMigrator(fileManager: fileManager)
+
+        expect { try migrate.migrateToScene(project: self.project) }.to(throwError(ProjectMigratorError.filePathNotFound))
```

Comment:
Which filePath is not found in this test? In general the behavior should be as follows:

There are certain projects which do not contain an _images_ or _sounds_ folder or any screenshot files. However, these projects should also be migrated, so there should not be any error in that case. Please change the test and the implementation in order to do so.

PS: ideally, in case there is no _images_ or _sounds_ directory available, please create it in the scene folder. Thanks!

Reply:
@m-herold  so the function `migrateToScene(project:)` should only throw an error if `automatic_screenshot.png` file is not present and all other files/directory are optional.

should this be the approach ?

Reply:
When a project has not been started yet, then this file does not exist either (no error in this case please).